src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/FindUnwantedIslandsTask.cpp

```diff
@@ -0,0 +1,136 @@
+/**
+ * UGENE - Integrated Bioinformatics Tools.
+ * Copyright (C) 2008-2021 UniPro <ugene@unipro.ru>
+ * http://ugene.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#include "FindUnwantedIslandsTask.h"
+
+#include <U2Core/DNASequenceUtils.h>
+#include <U2Core/PrimerDimersFinder.h>
+#include <U2Core/PrimerStatistics.h>
+#include <U2Core/U2SafePoints.h>
+
+#include "utils/UnwantedConnectionsUtils.h"
+
+namespace U2 {
+
+FindUnwantedIslandsTask::FindUnwantedIslandsTask(const U2Region& _searchArea, int _possibleOverlap, const QByteArray& _sequence, bool _isComplement)
+    : Task(tr("Find Unwanted Islands Task"), TaskFlags_FOSCOE),
+      searchArea(_searchArea),
+      possibleOverlap(_possibleOverlap),
+      sequence(_sequence),
+      isComplement(_isComplement) {}
+
+void FindUnwantedIslandsTask::run() {
+    taskLog.details(tr("Searching of unwanted islands and areas between them "
+                       "in the region \"%1\" (+ %2 nucleotides to the %3, deep into the amplified fragment) has been started")
+                        .arg(regionToString(searchArea))
+                        .arg(possibleOverlap)
+                        .arg(isComplement ? tr("left") : tr("right")));
+    taskLog.details(tr("The following unwanted parameters are used. Free Gibbs energy: %1 kj/mol, "
+                       "melting temperature: %2 C, maximum dimer length: %3 nt")
+                       .arg(UNWANTED_DELTA_G).arg(UNWANTED_MELTING_TEMPERATURE).arg(UNWANTED_MAX_LENGTH));
+    /**
+     * Index of the left nucleotide in the searching area.
+     */
+    int leftNucleotide = searchArea.startPos;
+    /**
+     * Index of the right nucleotide in the searching area.
+     * @settings.overlapLength.maxValue is the area extending deep into the amplified fragment.
+     */
+    const int rightNucleotide = searchArea.endPos() + possibleOverlap;
```

Comment:
EndPos is used here, but here https://github.com/ugeneunipro/ugene/blob/771b46aac08a5d4832a746b87cac3dd4cbe00383/src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/PCRPrimerDesignForDNAAssemblyTask.cpp#L108 endPos-1 is used.